Raygun4UWP/RUM/RUMService.cs

```diff
@@ -0,0 +1,320 @@
+
+      if (Application.Current != null)
+      {
+        Application.Current.Resuming += CurrentOnResuming;
+        Application.Current.Suspending += CurrentOnSuspending;
+      }
+
+      SendSessionStartEventInternalAsync();
+
+      _isEnabled = true;
+    }
+
+    public void Disable()
+    {
+      if (Application.Current != null)
+      {
+        Application.Current.Resuming -= CurrentOnResuming;
+        Application.Current.Suspending -= CurrentOnSuspending;
+      }
+
+      _isEnabled = false;
+    }
+
+    public async Task<RaygunRUMMessage> SendSessionStartEventAsync()
+    {
+      RaygunRUMMessage sessionStartMessage = null;
+
+      try
+      {
+        await SendSessionEndEventAsync();
+
+        _sessionId = Guid.NewGuid().ToString();
+
+        sessionStartMessage = BuildSessionEventMessage(RaygunRUMEventType.SessionStart, _sessionId);
+
+        await SendRUMMessageAsync(sessionStartMessage);
+      }
+      catch (Exception ex)
+      {
+        Debug.WriteLine($"Error sending session-start event to Raygun: {ex.Message}");
+      }
+
+      return sessionStartMessage;
+    }
+
+    public async Task<RaygunRUMMessage> SendSessionEndEventAsync()
+    {
+      RaygunRUMMessage sessionEndMessage = null;
+
+      try
+      {
+        if (_sessionId != null)
+        {
+          sessionEndMessage = BuildSessionEventMessage(RaygunRUMEventType.SessionEnd, _sessionId);
+
+          _sessionId = null;
+
+          await SendRUMMessageAsync(sessionEndMessage);
+        }
+      }
+      catch (Exception ex)
+      {
+        Debug.WriteLine($"Error sending session-end event to Raygun: {ex.Message}");
+      }
+
+      return sessionEndMessage;
+    }
+
+    public async Task<RaygunRUMMessage> SendSessionTimingEventAsync(RaygunRUMEventTimingType type, string name, long milliseconds)
+    {
+      RaygunRUMMessage sessionTimingMessage = null;
+
+      try
+      {
+        if (_sessionId == null)
+        {
+          await SendSessionStartEventAsync();
+        }
+
+        sessionTimingMessage = BuildSessionEventMessage(RaygunRUMEventType.Timing, _sessionId);
+
+        var data = new RaygunRUMTimingData[]
+        {
+          new RaygunRUMTimingData
+          {
+            Name = name,
+            Timing = new RaygunRUMTimingInfo
+            {
+              Type = type,
+              Duration = milliseconds
+            }
+          }
+        };
+
+        string dataPayload = JsonConvert.SerializeObject(data, HttpService.SERIALIZATION_SETTINGS);
+
+        sessionTimingMessage.EventData[0].Data = dataPayload;
+
+        await SendRUMMessageAsync(sessionTimingMessage);
+      }
+      catch (Exception ex)
+      {
+        Debug.WriteLine($"Error sending RUM timing event to Raygun: {ex.Message}");
+      }
+
+      return sessionTimingMessage;
+    }
+
+    public void ListenToNavigation(UIElement element)
+    {
+      if (element is Frame frame)
+      {
+        frame.Navigating += Frame_OnNavigating;

```

Comment:
Correct - the frame loads once, and we listen to this in order to send an event that specifies the initial page that the session starts on. A navigated event does not trigger in the loaded case - only on subsequent navigations.

Indeed you could have multiple frames loading at once - e.g. if you had two frames on the same page, which loads different content at the same time. Typically you would not attach RUM listening to such sibling frames, because a session should list pages that the user is currently on - one at a time. I can quickly look into improving this though in case it does get used in that way.